pkg/provider/provider.go

```diff
@@ -37,6 +37,32 @@ type CarbonIntensity struct {
 	Value         float64   `json:"value"`
 }
 
+type Details struct {
+	Name string
+	URL  string
+}
```

Comment:
This Details struct is for the metadata. We can add more fields like a description if we need them?